### PR TITLE
1829, SR completed, Tiles Maps, Corp

### PR DESCRIPTION
**Pull request by volker18xxdev**
I've changed the structure to buy_sell_par_shares_companies with grouping the Corporations.

### CODE REVIEW

**Comment by volker18xxdev** [comment]
last corrections where necessary due to some corrections in a 1835 file.

File: lib/engine/game/g_1829/entities.rb [hunk 1]
```diff
@@ -221,8 +328,50 @@ module Entities
...
+          'LYR' => 58,
+          'SECR' => 56,
+        }.freeze
+
+        REQUIRED_TRAIN = {
+        }.freeze
+
+        # combining is based on http://fwtwr.com/fwtwr/18xx/1829/privates.asp
+        def game_companies
+          comps = []
```
Comment (tobymao) [inline lib/engine/game/g_1829/entities.rb:365]: def game_companies
  @units[1] ? UNIT1_COMPANIES.dup : []

same for corporations

File: lib/engine/game/g_1829/game.rb [hunk 2]
```diff
@@ -88,102 +66,640 @@ class Game < Game::Base
...
+          add_train_list(trains, { '3' => -4, '4' => -3, '5' => -3 }) if @players.size == 2
+          add_train_list(trains, { '4' => -2, '5' => -2 }) if @players.size == 3
+          add_train_list(trains, { '4' => -1, '5' => -1 }) if @players.size == 4
+
+          fix_train_availables(trains)
+
+          trains
+        end
+
+        CURRENCY_FORMAT_STR = '£%d'
```
Comment (tobymao) [inline lib/engine/game/g_1829/game.rb:209]: put this on the top

**Comment by tobymao** [comment]
are you just copying large chunks of 1825 in this PR? if so, it's not really acceptable

**Comment by volker18xxdev** [comment]
i've asked Steve, and he gave permission to use his work. Map, Tiles, trains, Constants were done by myself. My first attemp was not successfull, because i need the grouping of corporations. I'd have to change to Steve's work cause of grouping the corporations and handling of combined trains.

**Comment by tobymao** [comment]
It’s not a matter of asking permission to copy but that copy and pasting
large chunks of code is bad programming practice.

**Comment by volker18xxdev** [comment]
that's true, but why create a good programmed code again? Other programmers also copied a complete game and then modified it.

**Comment by tobymao** [comment]
i don't have the time to thoroughly vet everything. we need to maintain this project with almost 40 games now, good code is a requirement.

**Comment by tobymao** [comment]
the code you copied is bad and has a major bug, @roseundy has fixed this, but since you copied and pasted it, it won't come through, i'm going to close this PR

**Comment by volker18xxdev** [comment]
ok, no more contribution from my side.

**Comment by tobymao** [comment]
i apologize if i've upset you but, i'm trying to keep the level of the code quality high so that we can maintain it for many years to come

**Comment by volker18xxdev** [comment]
You are the gatekeeper, it's ok. Please use the same scale for all. i.e. the constants are in 1825 not in the beginning, entities/Maps were in some games in the game.rb. It seems, that you make a difference between apprentices like me and old fellow programmers.

**Pull request closed without merging**